src/periodic.jl

```diff
@@ -0,0 +1,85 @@
+const PeriodicFunction = Union{typeof(cos), typeof(sin)}
+const PeriodicParameter = Union{Real,Period}
+
+"""
+    Periodic(f, period, phase_shift) <: Transform
```

Comment:
```suggestion
    Periodic{T}(f, period::T, phase_shift::T) <: Transform
```

might look too cluttered